### DISCUSSION TITLE
nav-item partial throws 'undefined variable' exception when building with a new Mac

### DISCUSSION THREAD

**Discussion by ms2art** [squidfunk/mkdocs-material#6065] [Q&A]
@squidfunk, wondering if you could speculate on this? 

My colleague is seeing a problem that I think is caused by something in the environment of his new Mac. MkDocs and the Material theme are set up properly, along with the required supporting software, but when he attempts a build that uses Martin Donath's nav-item.html partial, the build fails with a jinja2 exception at the end of the trace. Last few lines of trace:

File "/Users/theme_overrides/partials/nav-item.html", line 154, in top-level template code
navigation items #} {{ render(nav_item, path, level) }}
^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/jinja2/runtime.py", line 777, in _invoke
rv = self._func(*arguments)
^^^^^^^^^^^^^^^^^^^^^^
File "/Users/theme_overrides/partials/nav-item.html", line 19, in template
#} {% set class = "md-nav__item" %} {% if nav_item.active %} {% set class =
^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/jinja2/environment.py", line 485, in getattr
return getattr(obj, attribute)
^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'nav_item' is undefined

No output is generated, so probably the error occurs and stops the build on the first attempt to use the partial. This partial is crucial for us, because it reduces the size of the output HTML by showing only relevant portions of the TOC for each topic. The build works fine for me, and used to work on my colleague's old Mac, and we're using the same markdown file set as we always have. It feels like this must be a problem with the environment, and Jinja is the part complaining. Could there be some problem with the Jinja?  

Full trace and partial attached.

[jinja2-trace.txt](https://github.com/squidfunk/mkdocs-material/files/12702291/jinja2-trace.txt)

[nav-item.html.txt](https://github.com/squidfunk/mkdocs-material/files/12702295/nav-item.html.txt)

**Comment by squidfunk** [comment]
Looks like it's related to your overrides. I'd try to remove them and see if the error persists. If it's gone, check the diff between your overridden partial and the original.

**Reply by ms2art** [reply]
Yes, the problem is definitely caused by the nav-item.html partial. Can you direct me to the original nav-item.html partial? I think you wrote that, right?

**Reply by squidfunk** [reply]
It's quite easy to find it by using GitHub search ☺️

https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/nav-item.html

**Reply by ms2art** [reply]
Thanks. For your possible interest, we were able to get the partial to work by explicitly installing mkdocs-material 9.1.18, instead of the latest version. I'm not sure that's very helpful to you, because we are using a version of nav-item.html that's somewhat different from what you have posted. I don't have time to experiment more right now, but if you're interested and I get a breather, I can try to come back to it. It's odd that one version of mkdocs-material saw a variable as undefined and another didn't.

**Reply by squidfunk** [reply]
The partials may change from release to release. Sometimes we fix bugs, sometimes we add features, both of which might incur changes to partials. We're not considering partials part of the public API surface, which is why you need to check between releases if partials changed. We've recently started to mention changes in partials in our release notes, to make it simpler when upgrading.
